src/AppInstallerCLICore/ConfigurationSetProcessorFactoryRemoting.cpp

```diff
@@ -130,32 +109,126 @@ namespace AppInstaller::CLI::ConfigurationRemoting
+            IConfigurationSetProcessorFactory m_factory;
+            HRESULT m_result = S_OK;
+            wil::unique_event m_initEvent;
+        };
+
+        // Represents a remote factory object that was created from a specific process.
+        struct RemoteFactory : winrt::implements<RemoteFactory, IConfigurationSetProcessorFactory, SetProcessorFactory::IPwshConfigurationSetProcessorFactoryProperties, winrt::cloaked<WinRT::ILifetimeWatcher>>, WinRT::LifetimeWatcherBase
+        {
+            RemoteFactory(bool useRunAs, const std::string& properties, const std::string& restrictions)
+            {
+                AICLI_LOG(Config, Verbose, << "Launching process for configuration processing...");
 
-                THROW_HR_IF(E_NOT_SUFFICIENT_BUFFER, mappedMemory->FactorySize == 0);
-                THROW_HR_IF(E_NOT_SUFFICIENT_BUFFER, mappedMemory->FactorySize > details::MappedMemoryValue::MaxFactorySize());
+                // Create our callback and marshal it
+                auto callback = winrt::make_self<RemoteFactoryCallback>();
 
                 wil::com_ptr<IStream> stream;
                 THROW_IF_FAILED(CreateStreamOnHGlobal(nullptr, TRUE, &stream));
-                THROW_IF_FAILED(stream->Write(mappedMemory->FactoryObject, mappedMemory->FactorySize, nullptr));
+
+                THROW_IF_FAILED(CoMarshalInterface(stream.get(), winrt::guid_of<IConfigurationStatics>(), reinterpret_cast<::IUnknown*>(winrt::get_abi(callback.as<IConfigurationStatics>())), MSHCTX_LOCAL, nullptr, MSHLFLAGS_NORMAL));
+
+                ULARGE_INTEGER streamSize{};
+                THROW_IF_FAILED(stream->Seek({}, STREAM_SEEK_CUR, &streamSize));
+
+                ULONG bufferSize = static_cast<ULONG>(streamSize.QuadPart);
+                std::vector<uint8_t> buffer;
+                buffer.resize(bufferSize);
+
                 THROW_IF_FAILED(stream->Seek({}, STREAM_SEEK_SET, nullptr));
+                ULONG bytesRead = 0;
+                THROW_IF_FAILED(stream->Read(&buffer[0], bufferSize, &bytesRead));
+                THROW_HR_IF(E_UNEXPECTED, bytesRead != bufferSize);
+
+                std::wstring marshalledCallback = Utility::ConvertToUTF16(Utility::ConvertToHexString(buffer));
+
+                // Create the event that the remote process will wait on to keep the object alive.
+                std::wstring completionEventName = Utility::CreateNewGuidNameWString();
+                m_completionEvent.create(wil::EventOptions::None, completionEventName.c_str());
+                auto completeEventIfFailureDuringConstruction = wil::scope_exit([&]() { m_completionEvent.SetEvent(); });
 
-                wil::com_ptr<::IUnknown> output;
-                THROW_IF_FAILED(CoUnmarshalInterface(stream.get(), winrt::guid_of<IConfigurationSetProcessorFactory>(), reinterpret_cast<void**>(&output)));
+                // Arguments are:
+                // server.exe <marshalled callback object> <completion event name> <this process id>

```

Comment:
This had to change completely to support the elevation scenario, but both standard activation and the elevated activation are the same.